arch/xtensa/Kconfig

```diff
@@ -128,15 +128,23 @@ config XTENSA_USE_SEPARATE_IMEM
 		This separate heap will be part of the internal DRAM.  It starts right after .data
 		and ends at the configurable size given by the OPTION XTENSA_IMEM_REGION_SIZE.
 
+config XTENSA_IMEM_MAXIMIZE_HEAP_REGION
+	bool "Use a maximum separate heap for internal memory"
+	default n
+	depends on XTENSA_IMM_USE_SEPARATE_HEAP
+	help
+		This separate heap will be part of the internal DRAM.  It starts right after .data
+		and ends at the (HEAP_REGION1_END - HEAP_REGION_OFFSET).
+
 config XTENSA_IMEM_REGION_SIZE
 	hex "DRAM region size for internal use"
-	depends on XTENSA_USE_SEPARATE_IMEM
+	depends on XTENSA_IMM_USE_SEPARATE_HEAP && !XTENSA_IMEM_MAXIMIZE_HEAP_REGION
```

Comment:
are IMM and IMEM separate concepts?

Reply:
No, they are the same concepts, both means internal memory, I just have referred to the above @gustavonihei suggestion,   it's OK to use XTENSA_IMEM_USE_SEPARATE_HEAP

Reply:
it's better to be consistent at least within Kconfig. this seems like the only place IMM is used. @gustavonihei how do you think?

Reply:
Yes, let's please use the same abbreviation everywhere.

Reply:
I agree, it really improves the consistency.

Reply:
Done